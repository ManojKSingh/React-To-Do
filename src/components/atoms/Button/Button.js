import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
    background-color: ${props => props.theme.button.bgColor};
    color: ${props => props.theme.button.textColor};
    font-weight: bold;
    padding: 10px;
    margin: 0 5px;
    border: 2px solid ${props => props.theme.borderColor};
    border-radius: 4px;
    ${props =>
        props.styleType === 'remove'
        ? `
            padding: 2px 6px;
            border-radius: 50%;
            margin-left: auto;
            width: 22px;
            height: 22px;
            line-height: 0;
        `:``}
`;

const Button = ({children, ...otherProps}) => (
    <StyledButton {...otherProps}>
        {children}
    </StyledButton>
);

export default Button;
