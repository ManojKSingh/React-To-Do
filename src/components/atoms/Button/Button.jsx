import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledButton = styled.button`
    background-color: ${props => props.theme.button.bgColor};
    color: ${props => props.theme.button.textColor};
    font-weight: bold;
    padding: 10px;
    margin: 0 5px;
    border: 2px solid ${props => props.theme.borderColor};
    border-radius: 4px;
    ${props =>
        props.styleType === 'remove'
        ? `
            padding: 2px 6px;
            border-radius: 50%;
            margin-left: auto;
            width: 22px;
            height: 22px;
            line-height: 0;
        `:``}
`;

StyledButton.displayName = 'button';

const Button = ({children, onClick, ...otherProps}) => (
    <StyledButton onClick={onClick} {...otherProps}>
        {children}
    </StyledButton>
);

Button.defaultProps = {
  styleType: 'primary'
};

Button.propTypes = {
  onClick: PropTypes.func,
  styleType: PropTypes.oneOf(['primary', 'remove'])
};

export { Button };
