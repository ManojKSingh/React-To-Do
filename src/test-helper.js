import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import themes from '../src/themes';

const theme = themes['light'];

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}



