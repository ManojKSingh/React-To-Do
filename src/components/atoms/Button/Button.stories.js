import React from 'react';
import { action } from '@storybook/addon-actions';
import Button from './Button';

export default {
  title: 'Atom/Button',
  component: Button,
};

export const Text = () => <Button onClick={action('clicked')}>Hello Button</Button>;

export const Remove = () => <Button styleType='remove' onClick={action('clicked')}>x</Button>;
