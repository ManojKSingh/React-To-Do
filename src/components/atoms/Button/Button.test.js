import React from 'react';
import { renderWithTheme } from '../../../test-helper';
import { Button } from './Button';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Button/>', () => {
  it('renders correctly default button', () => {
    const tree = renderWithTheme(<Button>Add</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly remove button', () => {
    const tree = renderWithTheme(<Button styleType='remove'>x</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Test click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>));
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

