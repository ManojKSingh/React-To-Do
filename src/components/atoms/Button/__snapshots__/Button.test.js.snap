// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders correctly default button 1`] = `
<button
  className="sc-AxjAm kdpLzI"
>
  Add
</button>
`;

exports[`<Button/> renders correctly remove button 1`] = `
<button
  className="sc-AxjAm eiAZdM"
>
  x
</button>
`;
